diff -up ecl-13.5.1/src/aclocal.m4.end_of_line ecl-13.5.1/src/aclocal.m4
--- ecl-13.5.1/src/aclocal.m4.end_of_line	2014-05-18 15:11:30.019929425 -0500
+++ ecl-13.5.1/src/aclocal.m4	2014-05-18 15:12:54.630035302 -0500
@@ -711,7 +711,7 @@ int main() {
   fclose(f);
   f = fopen("conftestval","w");
   if (f == NULL) exit(1);
-  fprintf(f, output);
+  fprintf(f, "%s", output);
   fclose(f);
   exit(0);
 }
diff -up ecl-13.5.1/src/configure.end_of_line ecl-13.5.1/src/configure
--- ecl-13.5.1/src/configure.end_of_line	2014-05-18 15:11:30.022929393 -0500
+++ ecl-13.5.1/src/configure	2014-05-18 15:12:58.554993824 -0500
@@ -7680,7 +7680,7 @@ int main() {
   fclose(f);
   f = fopen("conftestval","w");
   if (f == NULL) exit(1);
-  fprintf(f, output);
+  fprintf(f, "%s", output);
   fclose(f);
   exit(0);
 }
