--- ecl-0.9h/src/gc/include/gc_cpp.h.gcc41	2006-03-15 17:11:48.000000000 +0100
+++ ecl-0.9h/src/gc/include/gc_cpp.h	2006-03-15 17:12:13.000000000 +0100
@@ -180,7 +180,7 @@
     inline void* operator new[]( size_t size, void *p );
     inline void operator delete[]( void* obj );
 #   ifdef GC_PLACEMENT_DELETE
-      inline void gc::operator delete[]( void*, void* );
+      inline void operator delete[]( void*, void* );
 #   endif
 #endif /* GC_OPERATOR_NEW_ARRAY */
     };    
