--- ./src/configure.in.orig	2012-07-24 10:51:53.000000000 -0600
+++ ./src/configure.in	2012-07-27 16:53:23.422951620 -0600
@@ -722,7 +722,6 @@ if test ! ${opcode8} = "no" ; then
 fi
 
 ECL_GCC_BACKTRACE
-ECL_LIBATOMIC_OPS
 
 dnl ----------------------------------------------------------------------
 dnl Lisp library and libecl features
--- ./src/Makefile.in.orig	2012-07-24 10:51:53.000000000 -0600
+++ ./src/Makefile.in	2012-07-27 16:53:05.966979669 -0600
@@ -123,11 +123,8 @@ bin/ecl-config: bin/ecl-config.pre
 	  mv @LIBPREFIX@ffi.@LIBEXT@ @LIBPREFIX@eclffi.@LIBEXT@; \
 	fi
 @LIBPREFIX@eclatomic.@LIBEXT@:
-	if (echo $(SUBDIRS) | grep atomic); then \
-	  cd atomic && $(MAKE) && $(MAKE) install && cd .. && \
-	  $(RM) -rf share ecl/atomic_ops_stack.h ecl/atomic_ops_malloc.h *atomic*gpl* && \
-	  mv @LIBPREFIX@atomic_ops.@LIBEXT@ @LIBPREFIX@eclatomic.@LIBEXT@; \
-	fi
+	cp -p ${libdir}/libatomic_ops.a $@
+	ln -s /usr/include/atomic_ops.h ecl/atomic_ops.h
 @LIBPREFIX@eclgmp.@LIBEXT@:
 	if (echo $(SUBDIRS) | grep gmp); then \
 	  cd gmp && $(MAKE) install && \
--- ./src/configure.orig	2012-07-24 10:51:53.000000000 -0600
+++ ./src/configure	2012-07-27 16:55:15.745769285 -0600
@@ -14248,23 +14248,6 @@ fi
 
 fi
 
-
-if test "x${enable_threads}" != "xno"; then
-  test -d atomic || mkdir atomic
-  (destdir=`${PWDCMD}`; cd atomic && CC="${CC} ${PICFLAG}" \
-   $srcdir/${ECL_GC_DIR}/libatomic*/configure --disable-shared --prefix=${destdir} \
-	--infodir=${destdir}/doc --includedir=${destdir}/ecl --with-pic \
-        --libdir=${destdir} --build=${build_alias} --host=${host_alias} \
-        CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" CPPFLAGS="$CPPFLAGS" CC="${CC} \
-        ${PICFLAG}")
-  SUBDIRS="${SUBDIRS} atomic"
-  CORE_LIBS="-leclatomic ${CORE_LIBS}"
-  if test "${enable_shared}" = "no"; then
-    LIBRARIES="${LIBRARIES} ${LIBPREFIX}eclatomic.${LIBEXT}"
-  fi
-fi
-
-
 if test ${enable_shared} = "yes"; then
 
 cat >>confdefs.h <<\_ACEOF
