--- src/configure.orig	2010-02-13 12:04:32.000000000 -0700
+++ src/configure	2010-02-24 12:36:49.949078016 -0700
@@ -5823,18 +5823,13 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
-#ifdef HAVE_SEMAPHORE_H
+#include <stdlib.h>
 #include <semaphore.h>
 int main() {
   sem_t aux;
   if (sem_init(&aux, 0, 0))
     exit(1);
   exit(0);
-#else
-int main() {
-  exit(1);
-}
-#endif
 }
 _ACEOF
 rm -f conftest$ac_exeext
