--- ./src/c/main.d.orig	2015-08-24 00:41:27.000000000 -0600
+++ ./src/c/main.d	2015-10-03 16:05:00.909971493 -0600
@@ -76,7 +76,11 @@ cl_fixnum ecl_option_values[ECL_OPT_LIMI
         1,              /* ECL_OPT_TRAP_SIGPIPE */
         1,              /* ECL_OPT_TRAP_SIGCHLD */
         1,              /* ECL_OPT_TRAP_INTERRUPT_SIGNAL */
+#ifdef GBC_BOEHM_GENGC
+        0,              /* ECL_OPT_SIGNAL_HANDLING_THREAD */
+#else
         1,              /* ECL_OPT_SIGNAL_HANDLING_THREAD */
+#endif
         16,             /* ECL_OPT_SIGNAL_QUEUE_SIZE */
         0,              /* ECL_OPT_BOOTED */
         8192,           /* ECL_OPT_BIND_STACK_SIZE */
@@ -498,11 +502,11 @@ cl_boot(int argc, char **argv)
                 return 1;
         }
 
-        /*ecl_set_option(ECL_OPT_SIGNAL_HANDLING_THREAD, 0);*/
-
 #if !defined(GBC_BOEHM)
         setbuf(stdin,  stdin_buf);
         setbuf(stdout, stdout_buf);
+#else
+        ecl_set_option(ECL_OPT_SIGNAL_HANDLING_THREAD, 0);
 #endif
 
         ARGC = argc;
