diff -up ecl-9.6.1/src/clx/dependent.lisp.mp ecl-9.6.1/src/clx/dependent.lisp
--- ecl-9.6.1/src/clx/dependent.lisp.mp	2009-06-16 21:47:04.000000000 +0200
+++ ecl-9.6.1/src/clx/dependent.lisp	2009-06-16 21:47:24.000000000 +0200
@@ -1321,7 +1321,7 @@
 
 #+(and ecl threads)
 (defun current-process ()
-  (mp:current-process))
+  mp:*current-process*)
 
 ;;; WITHOUT-INTERRUPTS -- provide for atomic operations.
 
