--- ./src/c/Makefile.in.orig	2015-08-24 00:41:27.000000000 -0600
+++ ./src/c/Makefile.in	2015-10-03 16:10:29.941777969 -0600
@@ -80,6 +80,29 @@ all:	$(DPP) ../libeclmin.a ../cinit.o
 .d.o: $(DPP)
 	if test -f ../CROSS-DPP ; then ../CROSS-DPP $< $@.c ; else $(DPP) $< $@.c ; fi
 	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -o $@ $@.c
+# GCC doesn't understand #pragma STDC FENV_ACCESS ON, so we do this instead:
+numbers/atan.o: numbers/atan.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/cos.o: numbers/cos.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/cosh.o: numbers/cosh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/exp.o: numbers/exp.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/expt.o: numbers/expt.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/log.o: numbers/log.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/sin.o: numbers/sin.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/sinh.o: numbers/sinh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/sqrt.o: numbers/sqrt.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/tan.o: numbers/tan.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
+numbers/tanh.o: numbers/tanh.c
+	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -frounding-math -o $@ $<
 .c.o:
 	$(CC) -DECLDIR="\"@ecldir@\"" $(CFLAGS) -o $@ $<
 .d.s: $(HFILES)
