diff --git a/src/configure b/src/configure
index b24cfc441..ce4477f7b 100755
--- a/src/configure
+++ b/src/configure
@@ -11348,6 +11348,7 @@ LSP_FEATURES="(cons :wants-dlopen ${LSP_FEATURES})"
 else
   enable_soname=no
   LSP_LIBRARIES="${LIBPREFIX}ecl.${LIBEXT}"
+  CFLAGS="${CFLAGS} -fPIC"
   ECL_LDRPATH=''
 fi
 
diff --git a/src/configure.ac b/src/configure.ac
index 309c03ea0..91325a61f 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -804,6 +804,7 @@ if test ${enable_shared} = "yes"; then
 else
   enable_soname=no
   LSP_LIBRARIES="${LIBPREFIX}ecl.${LIBEXT}"
+  CFLAGS="${CFLAGS} -fPIC"
   ECL_LDRPATH=''
 fi
 
